OpenRA.Mods.Common/Traits/Buildings/RallyPoint.cs

```diff
@@ -73,22 +75,34 @@ public IEnumerable<IOrderTargeter> Orders
 		public Order IssueOrder(Actor self, IOrderTargeter order, Target target, bool queued)
 		{
 			if (order.OrderID == "SetRallyPoint")
-				return new Order(order.OrderID, self, false) { TargetLocation = self.World.Map.CellContaining(target.CenterPosition), SuppressVisualFeedback = true };
+				return new Order(order.OrderID, self, false) { TargetLocation = self.World.Map.CellContaining(target.CenterPosition), SuppressVisualFeedback = true,
+					ExtraData = ((RallyPointOrderTargeter)order).SetPrimaryBuilding ? ExtraDataSetPrimaryBuilding : 0 };
 
 			return null;
 		}
 
 		public void ResolveOrder(Actor self, Order order)
 		{
 			if (order.OrderString == "SetRallyPoint")
+			{
 				Location = order.TargetLocation;
+				if (order.ExtraData == ExtraDataSetPrimaryBuilding)
+				{
+					var primaryBuildingTrait = self.Trait<PrimaryBuilding>();
```

Comment:
This should be TraitOrDefault otherwise you crash when there is no PrimaryBuilding trait.